Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/utils/Analytics.java

```diff
@@ -0,0 +1,116 @@
+package org.dolphinemu.dolphinemu.utils;
+
+import android.app.AlertDialog;
+import android.content.Context;
+import android.content.IntentFilter;
+import android.content.SharedPreferences;
+import android.os.Build;
+import android.preference.PreferenceManager;
+import android.support.v4.content.LocalBroadcastManager;
+
+import com.android.volley.Request;
+import com.android.volley.toolbox.StringRequest;
+
+import org.dolphinemu.dolphinemu.R;
+import org.dolphinemu.dolphinemu.features.settings.model.Settings;
+import org.dolphinemu.dolphinemu.features.settings.utils.SettingsFile;
+import org.dolphinemu.dolphinemu.services.DirectoryInitializationService;
+
+public class Analytics
+{
+	private static DirectoryStateReceiver directoryStateReceiver;
+
+	private static final String analyticsAsked = Settings.SECTION_ANALYTICS + "_" + SettingsFile.KEY_ANALYTICS_PERMISSION_ASKED;
+	private static final String analyticsEnabled = Settings.SECTION_ANALYTICS + "_" + SettingsFile.KEY_ANALYTICS_ENABLED;
+
+	private static final String DEVICE_MANUFACTURER = "DEVICE_MANUFACTURER";
+	private static final String DEVICE_OS = "DEVICE_OS";
+	private static final String DEVICE_MODEL = "DEVICE_MODEL";
+
+	public static void checkAnalyticsInit(Context context)
+	{
+		SharedPreferences preferences = PreferenceManager.getDefaultSharedPreferences(context);
+		if (!preferences.getBoolean(analyticsAsked, false))
+		{
+			if (!DirectoryInitializationService.areDolphinDirectoriesReady())
+			{
+				// Wait for directories to get initialized
+				IntentFilter statusIntentFilter = new IntentFilter(
+						DirectoryInitializationService.BROADCAST_ACTION);
+
+				directoryStateReceiver = new DirectoryStateReceiver(directoryInitializationState ->
+				{
+					if (directoryInitializationState == DirectoryInitializationService.DirectoryInitializationState.DOLPHIN_DIRECTORIES_INITIALIZED)
+					{
+						LocalBroadcastManager.getInstance(context).unregisterReceiver(directoryStateReceiver);
+						directoryStateReceiver = null;
+						showMessage(context, preferences);
+					}
+				});
+				// Registers the DirectoryStateReceiver and its intent filters
+				LocalBroadcastManager.getInstance(context).registerReceiver(
+						directoryStateReceiver,
+						statusIntentFilter);
+			}
+			else
+				showMessage(context, preferences);
```

Comment:
Would you mind adding curlies here? Should be balanced since the `if` has them, even when the `else` doesn't need it.